Source/WebCore/platform/graphics/cocoa/VideoMediaSampleRenderer.mm

```diff
@@ -440,11 +450,15 @@ static OSStatus sampleCallback(CMBufferRef buffer, void* refcon)
 
     auto decodePromise = decompressionSession->decodeSample(sample.get(), displaying);
     m_isDecodingSample = true;
-    decodePromise->whenSettled(dispatcher(), [weakThis = ThreadSafeWeakPtr { *this }, this, displaying, flushId = flushId](auto&& result) {
+    decodePromise->whenSettled(dispatcher(), [weakThis = ThreadSafeWeakPtr { *this }, this, displaying, flushId = flushId, startTime = MonotonicTime::now()](auto&& result) {
         RefPtr protectedThis = weakThis.get();
         if (!protectedThis)
             return;
 
+#if LOG_DECODER_TIMINGS
+        auto now = MonotonicTime::now();
+        RELEASE_LOG_ERROR(Media, "decoding rate:%0.1fHz decoder rate:%0.1fHz compressed queue:%u decoded queue:%zu", 1.0f / Seconds { now - std::exchange(m_timeSinceLastDecode, now) }.value(), 1.0f / Seconds { now - startTime }.value(), m_compressedSampleQueueSize.load(), decodedSamplesCount());
+#endif
```

Comment:
Is this needed? If so, I would remove LOG_DECODER_TIMINGS and just use a LOG(Media...).
An alternative is to use `FrameRateMonitor`.

Reply:
LOG(Media) is only available on debug build AFAIK. And those logs are only of value with release build

Reply:
Added a new MediaPerformance logging category